### PR TITLE
Update stackdriver filter to fallback to default mesh id format when not specified

### PR DESCRIPTION
**Pull request by bianpengyuan**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: extensions/stackdriver/stackdriver.cc [hunk 1]
```diff
@@ -151,12 +151,45 @@ std::string getMonitoringEndpoint() {
...
+      !mesh_id_it->second.string_value().empty()) {
+    return;
+  }
+
+  // Insert or update mesh id to default mesh id if empty.
+  auto project_number = getProjectNumber();
+  auto* mesh_id_field =
+      (*node_metadata->mutable_fields())["MESH_ID"].mutable_string_value();
+  if (project_number.empty()) {
+    *mesh_id_field = "unknown";
```
Comment (mandarjog) [inline extensions/stackdriver/stackdriver.cc:187]: What is the difference between meshid = unknown and unset?

Comment (bianpengyuan) [reply extensions/stackdriver/stackdriver.cc]: Actually not much, I think both edge and metric handle this case well (if empty, set it to unknown). For log, it currently leaves unknown fields empty, so this is going to be make a difference there.

**Comment by bianpengyuan** [comment]
@elfinhe fyi.

File: extensions/stackdriver/stackdriver.cc [hunk 2]
```diff
@@ -151,12 +151,43 @@ std::string getMonitoringEndpoint() {
...
+  if (!getMessageValue({"node", "metadata"}, node_metadata)) {
+    return;
+  }
+  const auto mesh_id_it = node_metadata->fields().find("MESH_ID");
+  if (mesh_id_it != node_metadata->fields().end() &&
+      !mesh_id_it->second.string_value().empty()) {
+    return;
+  }
+
+  // Insert or update mesh id to default mesh id if empty.
```
Comment (gargnupur) [inline extensions/stackdriver/stackdriver.cc:182]: should the comment says "as it's empty"

with  "if empty", it feels it should check for it being empty below...

Comment (bianpengyuan) [reply extensions/stackdriver/stackdriver.cc]: Done, thanks!

File: extensions/stackdriver/stackdriver.cc [hunk 3]
```diff
@@ -151,12 +151,43 @@ std::string getMonitoringEndpoint() {
...
   request_info.tcp_received_bytes = 0;
 }
 
+// Get local node metadata. If mesh id is not filled or does not exist,
+// fall back to default format `proj-<project-number>`.
+void getLocalNodeMetadata(google::protobuf::Struct* node_metadata) {
+  if (!getMessageValue({"node", "metadata"}, node_metadata)) {
+    return;
+  }
+  const auto mesh_id_it = node_metadata->fields().find("MESH_ID");
```
Comment (elfinhe) [inline extensions/stackdriver/stackdriver.cc:176]: Is there any default value for the MESH_ID here?

File: extensions/stackdriver/stackdriver.cc [hunk 4]
```diff
@@ -151,12 +151,43 @@ std::string getMonitoringEndpoint() {
...
+  if (mesh_id_it != node_metadata->fields().end() &&
+      !mesh_id_it->second.string_value().empty()) {
```
Comment (elfinhe) [inline extensions/stackdriver/stackdriver.cc:178]: Today the meshid will always fallback to trust domain "<projectname>.svc.id.goog", which cannot be recognized by stackdriver.

we need to add one more line, 

```
// Today GCP Monitoring use "proj-<ProjectNumber>" as mesh ID, reuse input if it is a valid mesh ID.
 if (mesh_id_it != node_metadata->fields().end() &&
      !mesh_id_it->second.string_value().empty() &&
      mesh_id_it->second.string_value().<HAS_PREFIX>("proj-")) {
            return;
      }
```

Comment (mandarjog) [reply extensions/stackdriver/stackdriver.cc:178]: Great catch.
I question the useless default: svc.id.goog ?

File: extensions/stackdriver/stackdriver.cc (hunk 3 above)
Comment (bianpengyuan) [reply extensions/stackdriver/stackdriver.cc:176]: It will just be empty string if not set.

File: extensions/stackdriver/stackdriver.cc (hunk 4 above)
Comment (bianpengyuan) [reply extensions/stackdriver/stackdriver.cc]: Done.

**Pull request merged**